rubicon_ml/intake_rubicon/publish.py

```diff
@@ -47,3 +59,60 @@ def publish(
             f.write(catalog_yaml)
 
     return catalog_yaml
+
+
+"""Helper function to update exisiting intake catalog.
+
+    Parameters
+    ----------
+    base_catalog_filepath : str
+        the absolute or relative catalog filepath or S3 bucket
+        and key to log the generated YAML file to. S3 buckets
+        must be prepended with 's3://. Retrieved from the parameter
+        of the publish function. NOT optional
+    new experiments : list of rubicon_ml.client.experiment.Experiment
+        The new experiments to publish.
+    output_catalog_filepath : str, optional
+        absolute or relative filepath or S3 bucket
+        and key to log the generated YAML file to. (S3 buckets
+        must be prepended with 's3://) to  output the updated catalog into.
+        Default is None, which resolves to dumping updated catalog into
+        base_catalog_filepath path (primary use-case)
+
+    Returns
+    -------
+    str
+        The YAML string representation of the `intake` catalog
+        containing the experiments `experiments`.
+    """
+
+
+def update_catalog(base_catalog_filepath, new_experiments, output_filepath=None):
+
+    updated_catalog = {}
+    for experiment in new_experiments:
+        appended_experiment_catalog = {
+            "args": {
+                "experiment_id": experiment.id,
+                "project_name": experiment.project.name,
+                "urlpath": experiment.repository.root_dir,
+            },
+            "driver": "rubicon_ml_experiment",
+        }
+
+        experiment_catalog_name = f"experiment_{experiment.id.replace('-', '_')}"
+        updated_catalog[experiment_catalog_name] = appended_experiment_catalog
+
+    with fsspec.open(base_catalog_filepath, "r") as yamlfile:
+        curr_catalog = yaml.safe_load(yamlfile)
+
+        curr_catalog["sources"].update(updated_catalog)
+
+    resulting_filepath = base_catalog_filepath if not output_filepath else output_filepath
+    if curr_catalog:
```

Comment:
is this `if` statement necessary? `curr_catalog` will always not be None, right?